src/FormElementManager.php

```diff
@@ -321,9 +321,10 @@ public function configure(array $config)
      * createFromInvokable() will use these and pass them to the instance
      * constructor if not null and a non-empty array.
      *
-     * @param class-string<ElementInterface>|string $name Service name of plugin to retrieve.
+     * @template T of ElementInterface
+     * @param class-string<T>|string $name Service name of plugin to retrieve.
      * @param null|array<mixed> $options Options to use when creating the instance.
-     * @psalm-return ($name is class-string<ElementInterface> ? ElementInterface : mixed)
+     * @psalm-return ($name is class-string<T> ? T : mixed)
```

Comment:
```suggestion
     * @psalm-return ($name is class-string<ElementInterface> ? T : ElementInterface)
```

This does not work as expected. See https://github.com/laminas/laminas-inputfilter/pull/97#discussion_r1446307372 for more context

Also instead of mixed as fallback return `ElementInterface` as form element manager validates return type.

Add regular `@return` annotation for IDEs